e2e_playwright/mega_tester_app_test.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2025)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from playwright.sync_api import expect
+
+from e2e_playwright.conftest import IframedPage, wait_for_app_loaded, wait_for_app_run
+
+if TYPE_CHECKING:
+    from playwright.sync_api import ConsoleMessage, FrameLocator, Page
+
+
+def is_expected_error(msg: ConsoleMessage, browser_name: str, *, uses_csp: bool):
+    # Mapbox error is expected and should be ignored:
+    if (
+        msg.text == "Failed to load resource: net::ERR_CONNECTION_REFUSED"
+        and "events.mapbox.com" in msg.location["url"]
+    ):
+        return True
+
+    # There is an expected error with pydeck and firefox related to WebGL rendering
+    # This seems to be an issue with firefox used with playwright:
+    if msg.text == "deck: o is null undefined" and browser_name == "firefox":
+        return True
+
+    # TODO(lukasmasuch): Investigate why firefox is running into this eval issue:
+    if (
+        (
+            "settings blocked a JavaScript eval (script-src) from being executed"
+            in msg.text
+        )
+        and browser_name == "firefox"
+        and uses_csp
+    ):
+        return True
+
+    # TODO(lukasmasuch): Investigate why webkit is running into this blob: issue when
+    if (
+        msg.text == "Failed to load resource"
+        and "blob:http://localhost:" in msg.location["url"]
+        and browser_name == "webkit"
+        and uses_csp
+    ):
+        return True
```

Comment:
I cannot reproduce these errors with my local firefox/webkit browser. It might be only related to something specific when running this with Playwright.